wayland-protocols/ext-input-trigger-action-v1.xml

```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="ext_input_trigger_action_v1">
+  <copyright>
+    Copyright 2025 Canonical Limited
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE.
+  </copyright>
+
+  <description summary="Registration for global input actions">
+    This protocol allows for a Wayland clients to subscribe to input trigger
+    actions for "begin", "update, and "end" notifications.
+
+    ext_input_trigger_action_v1 and the associated ext_input_trigger_registration_v1
+    are based on the following discussion document:
+    https://docs.google.com/document/d/1Hj3BXkmW1quIEWqmN7xeJY29nSPfXP2bmoZvpOsEeJQ/
+  </description>
+
+  <interface name="ext_input_trigger_action_manager_v1" version="1">
+    <description summary="used to register interest in input actions">
+      This interface allows a Wayland client to register interest in
+      input trigger actions.
+    </description>
+
+    <request name="destroy" type="destructor">
+      <description summary="destroy input trigger activation manager object">
+        This informs the compositor that the input trigger activation manager
+        object will no longer be used. Existing objects created through this
+        interface remain valid.
+      </description>
+    </request>
+
+    <request name="get_input_trigger_action">
+      <description summary="register interest in input trigger action">
+        This request creates an input trigger action object.
+
+        The token should be one issued by the compositor, if not then the
+        compositor can raise a protocol error.
```

Comment:
This should be more prescriptive: the compositor *should* raise a protocol error, but which protocol error? None appear to be defined.

Reply:
> But in my opinion, it's valid here to raise a protocol error and terminate the client if they pass in an incorrect token.

Yes, but *what* protocol error? You can't just send “error”, it needs to be a particular thing (but there are no errors defined in this protocol).